public/app/features/explore/AddToDashboard/index.test.tsx

```diff
@@ -84,218 +48,114 @@ describe('Add to Dashboard Button', () => {
-      ...createEmptyQueryResponse(),
-      graphFrames: [new MutableDataFrame({ refId: 'A', fields: [] })],
+      userEvent.click(screen.getByRole<HTMLInputElement>('radio', { name: /existing dashboard/i }));
+      expect(screen.getByRole('combobox', { name: /dashboard/ })).toBeInTheDocument();
     });
 
-    await openModal();
-
-    userEvent.click(screen.getByRole('button', { name: /save and keep exploring/i }));
-
-    await waitForElementToBeRemoved(() => screen.queryByRole('dialog', { name: 'Add panel to dashboard' }));
-
-    expect(addToDashboardMock).toHaveBeenCalledWith(
-      expect.anything(),
-      expect.objectContaining({
-        panel: 'timeseries',
-      })
-    );
-  });
+    it('Does not submit if no dashboard is selected', async () => {
+      locationService.push = jest.fn();
 
-  it('Sets visualization to nodeGraph if there are node graph frames', async () => {
-    const queries: DataQuery[] = [{ refId: 'A' }];
-    setup(<AddToDashboard exploreId={ExploreId.left} />, queries, {
-      ...createEmptyQueryResponse(),
-      nodeGraphFrames: [new MutableDataFrame({ refId: 'A', fields: [] })],
-    });
+      setup(<AddToDashboard exploreId={ExploreId.left} />);
 
-    await openModal();
+      await openModal();
 
-    userEvent.click(screen.getByRole('button', { name: /save and keep exploring/i }));
+      userEvent.click(screen.getByRole<HTMLInputElement>('radio', { name: /existing dashboard/i }));
 
-    await waitForElementToBeRemoved(() => screen.queryByRole('dialog', { name: 'Add panel to dashboard' }));
+      userEvent.click(screen.getByRole('button', { name: /open$/i }));
+      await waitForAddToDashboardResponse();
 
-    expect(addToDashboardMock).toHaveBeenCalledWith(
-      expect.anything(),
-      expect.objectContaining({
-        panel: 'nodeGraph',
-      })
-    );
-  });
-
-  // trace view is not supported in dashboards, defaulting to table
-  it('Sets visualization to table if there are trace frames', async () => {
-    const queries: DataQuery[] = [{ refId: 'A' }];
-    setup(<AddToDashboard exploreId={ExploreId.left} />, queries, {
-      ...createEmptyQueryResponse(),
-      traceFrames: [new MutableDataFrame({ refId: 'A', fields: [] })],
+      expect(locationService.push).not.toHaveBeenCalled();
     });
 
-    await openModal();
-
-    userEvent.click(screen.getByRole('button', { name: /save and keep exploring/i }));
-
-    await waitForElementToBeRemoved(() => screen.queryByRole('dialog', { name: 'Add panel to dashboard' }));
+    describe('Navigate to correct dashboard when saving', () => {
+      it('Opens the selected dashboard in a new tab', async () => {
+        // TODO: implement
+      });
 
-    expect(addToDashboardMock).toHaveBeenCalledWith(
-      expect.anything(),
-      expect.objectContaining({
-        panel: 'table',
-      })
-    );
+      it('Navigates to the selected dashboard', async () => {
+        // TODO: implement
+      });
+    });
```

Comment:
I'm getting some errors regarding async state update related to AsyncSelect. I'll get back to these later